test/solver/ir_kernels.cpp

```diff
@@ -243,3 +243,37 @@ TEST_F(Ir, RichardsonApplyWithIterativeInnerSolverIsEquivalentToRef)
     // difference in IR.
     GKO_ASSERT_MTX_NEAR(d_x, x, r<value_type>::value * 200);
 }
+
+
+TEST_F(Ir, ApplyWithGivenHintIsEquivalentToRef)
+{
+    using input_hint = gko::solver::input_hint;
+    auto mtx = gko::share(gen_mtx(50, 50, 52));
+    auto b = gen_mtx(50, 3, 7);
+    auto d_mtx = gko::share(clone(exec, mtx));
+    auto d_b = clone(exec, b);
+    for (auto hint : {input_hint::given, input_hint::rhs, input_hint::zero}) {
```

Comment:
Maybe, create a `std::initializer_list<gko::solver::input_hint>` in the `solver` header, so this is not repeated in every single place and the enum is iterable through that?

Reply:
I think the iteration is only for test purposes, so I wouldn't like to introduce in public header